src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,210 @@
+  TouchableOpacity,
+  Easing,
+  StyleProp,
+  Text,
+  TextStyle,
+  Platform,
+  TouchableOpacityProps
+} from 'react-native';
+import dimen from '../../utils/dimen';
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  basicStyle: {
+    height: dimen.normalizeDimen(16),
+    width: dimen.normalizeDimen(16),
+    borderRadius: dimen.normalizeDimen(16 / 2),
+    backgroundColor: '#2C67BC'
+  },
+  eahcStyles: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    left: dimen.normalizeDimen(3)
+  },
+
+  eachStylesOff: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    right: dimen.normalizeDimen(3)
+  },
+  mainStyes: {
+    borderRadius: dimen.normalizeDimen(12),
+    backgroundColor: '#F5F5F5',
+    height: dimen.normalizeDimen(20),
+    width: dimen.normalizeDimen(42),
+    padding: dimen.normalizeDimen(2),
+    ...Platform.select({
+      ios: {
+        shadowColor: 'rgba(0,0,0,0.25)',
+        shadowOffset: {width: 0, height: 0},
+        shadowOpacity: dimen.normalizeDimen(1),
+        shadowRadius: dimen.normalizeDimen(1)
+      },
+      android: {
+        elevation: dimen.normalizeDimen(1)
+      }
+    })
+  },
+  labelLeft: {
+    marginRight: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  },
+  labelRight: {
+    marginLeft: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  }
+});
+
+interface ToggleSwitchProps {
+  value?: boolean;
+  labelLeft?: string;
+  labelRight?: string;
+  circleActiveColor?: string;
+  circleInActiveColor?: string;
+  backgroundActive?: string;
+  backgroundInactive?: string;
+  styleLabelLeft?: StyleProp<TextStyle>;
+  styleLabelRight?: StyleProp<TextStyle>;
+  activeTextColor?: string;
+  inactiveTextColor?: string;
+  containerStyle?: TouchableOpacityProps['style'];
+  labelOff?: string;
+  labelOn?: string;
+  onValueChange?: (value: boolean) => void;
+}
+
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  value = false,
+  onValueChange,
+  labelLeft,
+  labelRight,
+  circleActiveColor = '#00ADB5',
+  circleInActiveColor = '#2C67BC',
+  backgroundActive = '#F5F5F5',
+  backgroundInactive = '#F5F5F5',
+  styleLabelLeft,
+  styleLabelRight,
+  activeTextColor = '#00ADB5',
+  inactiveTextColor = '#2C67BC',
+  containerStyle,
+  labelOff = 'Off',
+  labelOn = 'On'
+}) => {
+  const positionButton = React.useMemo(() => new Animated.Value(0), []);
+
+  const startAnimToOff = () => {
+    Animated.timing(positionButton, {
+      toValue: 0,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
+
+  const startAnimToOn = () => {
+    Animated.timing(positionButton, {
+      toValue: 1,
+      duration: 150,
+      easing: Easing.ease,
+      useNativeDriver: false
+    }).start();
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The animations for switching the toggle on and off are well implemented using React's `useMemo` and `Animated.timing`. However, you might want to consider moving these functions outside of the component to avoid redefining them on every render. 

```diff
+ const startAnimToOff = (positionButton: Animated.Value) => {
+   Animated.timing(positionButton, {
+     toValue: 0,
+     duration: 150,
+     easing: Easing.ease,
+     useNativeDriver: false
+   }).start();
+ };
+
+ const startAnimToOn = (positionButton: Animated.Value) => {
+   Animated.timing(positionButton, {
+     toValue: 1,
+     duration: 150,
+     easing: Easing.ease,
+     useNativeDriver: false
+   }).start();
+ };

- const startAnimToOff = () => {
-   Animated.timing(positionButton, {
-     toValue: 0,
-     duration: 150,
-     easing: Easing.ease,
-     useNativeDriver: false
-   }).start();
- };
-
- const startAnimToOn = () => {
-   Animated.timing(positionButton, {
-     toValue: 1,
-     duration: 150,
-     easing: Easing.ease,
-     useNativeDriver: false
-   }).start();
- };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->